### PR TITLE
Object3D: granular control of matrix updates with matrixWorldAutoUpdate

### PR DESCRIPTION
**Pull request by CodyJasonBennett**
Related issue: #21387

**Description**

Currently, all scene objects have their matrices updated in `Object3D#updateWorldMatrix` or `Object3D#updateMatrixWorld` if somehow added to the scene. You can toggle this behavior with `Scene.autoUpdate`, but there's no way to let other objects do the same at any point in the scene graph.

This PR moves the `Scene.autoUpdate` property to `Object3D.matrixWorldAutoUpdate` to enable object-level opt-out of matrix updates in `Object3D#updateWorldMatrix` and `Object3D#updateMatrixWorld`. This behavior extends to matrix updates to cameras and scene objects from renderers.

### CODE REVIEW

**Comment by CodyJasonBennett** [comment]
Should we have this check against `camera.autoUpdate` instead?

https://github.com/mrdoob/three.js/blob/fff7be399aba68375b3b65eef809627110e36413/src/renderers/WebGLRenderer.js#L921-L927

**Comment by AlaricBaraou** [comment]
Just saw that LOD objects extend Object3D and already use the autoUpdate property. 

I thought this might be worth mentioning

File: src/core/Object3D.js [hunk 1]
```diff
@@ -921,6 +933,7 @@ class Object3D extends EventDispatcher {
 
 Object3D.DefaultUp = new Vector3( 0, 1, 0 );
 Object3D.DefaultMatrixAutoUpdate = true;
+Object3D.DefaultAutoUpdate = true;
```
Comment (mrdoob) [inline src/core/Object3D.js:936]: For consistency...

Should we just call it `DefaultMatrixWorldAutoUpdate`? ... and `this.matixWorldAutoUpdate`?

**Comment by mrdoob** [comment]
For consistency...

Should we just name it `DefaultMatrixWorldAutoUpdate`? ... and `matixWorldAutoUpdate`?

**Comment by CodyJasonBennett** [comment]
I'd much prefer that to just `autoUpdate`. From a casual perspective, I wouldn't otherwise be able to guess what it does.

**Comment by CodyJasonBennett** [comment]
Seems I found a way to break DeepScan when reverting changes to examples/js. Force-pushing didn't help either.

**Comment by mrdoob** [comment]
Thanks!

**Comment by mrdoob** [comment]
A few tweaks: 7ca0bb679cc4b8d4f590af475cf4e550e07724fe 757dadf2333b5a62f75b1263acf0007abe2bb5b3 c6e93866b79044690ed1ddf9279d9062dac2059d

**Pull request merged**